x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -147,17 +153,33 @@ export async function generatePackageInfoFromArchiveBuffer(
   };
 }
 
-function parseAndVerifyArchive(paths: string[]): ArchivePackage {
+/*
+This is a util function for verifying packages from a directory not an archive.
+It is only to be called from test scripts.
+*/
+export async function _generatePackageInfoFromPaths(
+  paths: string[],
+  topLevelDir: string
+): Promise<ArchivePackage> {
+  await Promise.all(
+    paths.map(async (filePath) => {
+      if (filePath.endsWith(MANIFEST_NAME)) MANIFESTS[filePath] = await readFileAsync(filePath);
+    })
+  );
+  return parseAndVerifyArchive(paths, topLevelDir);
+}
+
+function parseAndVerifyArchive(paths: string[], topLevelDirOverride?: string): ArchivePackage {
```

Comment:
I have added the ability to override the "top level directory" as the test packages do not follow the same file structure as when we unpack the archive.